### PR TITLE
Alexnet with Code

### PR DESCRIPTION
**Pull request by ash-01xor**
- Gives a small introduction followed about by the features and the code implementation .

### CODE REVIEW

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Alexnet
+
+# Introduction
+
+In this chapter we will take a look at AlexNet, the network that began the revolution of deep convolutional neural network Architechture that revolutionized the field
```
Comment (MKhalusova) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:5]: "revolution ... that revolutionized" is a bit repetitive. please rephrase.

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+of computer vision.It was introduced by Alex Krizhevsky, Ilya Sutskever, and Geoffrey Hinton in their 2012 paper [ImageNet Classification with Deep Convolutional Neural Networks](https://proceedings.neurips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf) and won the ImageNet Large Scale Visual Recognition Challenge (ILSVRC) in 2012.
+The ImageNet dataset consisted of more than 15 million high resolution images.
```
Comment (MKhalusova) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:7]: This last sentence feels somewhat out of context, or lacking in cohesive ties to the rest of the paragraph.

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+- AlexNet also introduced Local Response Normalization (LRN) layers, which helped to improve the generalization performance of the network.
+
+5.ReLU Activation:
+
+- AlexNet was one of the first architectures to effectively utilize the Rectified Linear Unit (ReLU) activation function, which helped to address the vanishing gradient problem and accelerated the training process.
+
+### Key features
+
+- 'ReLU' is used as an activation function than 'tanh'
```
Comment (MKhalusova) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:38]:
```suggestion
- 'ReLU' is used as an activation function as opposed to 'tanh'
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    - 'tanh' was the standard activation function which was used back in those days.
+    - The primary advantage than 'ReLU' offered was the fact it reduced training time and also help in helped to alleviating the vanishing gradient problem
```
Comment (MKhalusova) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:40]:
```suggestion
    - The primary advantage than 'ReLU' offered was the fact it reduced training time and also help helped to alleviate the vanishing gradient problem
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+- Usage of Dropout mechanism
+    - It is a regularization technique that helps to prevent overfitting in neural networks by randomly dropping out (setting to zero) a proportion of units (neurons) and their connections during the training process.
+    - Forces the network to learn more robust representation
+- Data Augmentation techniques such as flipping,jittering,cropping etc were performed.
```
Comment (MKhalusova) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:44]:
```suggestion
- Data Augmentation techniques such as flipping, jittering, cropping, etc. were performed.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    - The authors employed employed extensive data augmentation techniques to artificially increase the size and diversity of the training data.
```
Comment (MKhalusova) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:45]:
```suggestion
    - The authors employed extensive data augmentation techniques to artificially increase the size and diversity of the training data.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    - This helped in not only increasing the dataset size , but also improved generalization and acted as a form of regularization.
```
Comment (MKhalusova) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:46]:
```suggestion
    - This helped in not only increasing the dataset size, but also aided generalization and acted as a form of regularization.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+### Example implementation using Pytorch
+
+```
```
Comment (MKhalusova) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:50]:
```suggestion
```python
```

**Review by MKhalusova — COMMENTED** [review]
Thank you for adding this page, I left some suggestions.

**Comment by ATaylorAerospace** [comment]
@MKhalusova caught all the nits! I dont see anything else and will approve once you have made your updates @ash-01xor so we can merrge!

**Review by ATaylorAerospace — COMMENTED** [review]
@MKhalusova has caught all the nits so I am ready to help approve once you make your updates @ash-01xor so we can merge!

**Review by MKhalusova — APPROVED** [review]
Thank you for all the updates!

**Review by ATaylorAerospace — APPROVED** [review]
@ash-01xor Thanks for the fast updates

**Comment by ash-01xor** [comment]
Are any more changes required or can this PR me merged @MKhalusova

**Comment by ash-01xor** [comment]
> All good from my side, just the conflict that needs to be resolved.

Have resolved it @MKhalusova

**Comment by ATaylorAerospace** [comment]
@ash-01xor This is very close to merge but there is one more check that needs to be resolved. Can you check Heading and CodeBlock as indicated in thre Make Documentation check error

**Comment by ash-01xor** [comment]
> @ash-01xor This is very close to merge but there is one more check that needs to be resolved. Can you check Heading and CodeBlock as indicated in thre Make Documentation check error

Hey @ATaylorAerospace  i was trying to resolve it and push the commit , however unknown issues popped up and I'm unable to work on it. Can you resolve it ?

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 9]
```diff
@@ -0,0 +1,91 @@
+# Alexnet
+
+# Introduction
+
+In this chapter we will take a look at AlexNet, the network that began the revolution of deep convolutional neural network Architechture.
+of computer vision.It was introduced by Alex Krizhevsky, Ilya Sutskever, and Geoffrey Hinton in their 2012 paper [ImageNet Classification with Deep Convolutional Neural Networks](https://proceedings.neurips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf) and won the ImageNet Large Scale Visual Recognition Challenge (ILSVRC) in 2012.
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:6]:
```suggestion
In this chapter we will take a look at AlexNet, the network that began the revolution of Deep Convolutional Neural Network architectures.
It was introduced by Alex Krizhevsky, Ilya Sutskever, and Geoffrey Hinton in their 2012 paper [ImageNet Classification with Deep Convolutional Neural Networks](https://proceedings.neurips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf) and won the ImageNet Large Scale Visual Recognition Challenge (ILSVRC) in 2012.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+The ImageNet dataset consisted of more than 15 million high resolution images that posed a significant challenge to traditional computer vision algorithms, as it demanded an unprecedented level of visual understanding and classification capabilities.
+AlexNet's triumph in the ILSVRC marked a turning point in the field of computer vision, demonstrating the superior performance of deep convolutional neural networks over conventional machine learning techniques.
+
+### Architechture of Alexnet
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:10]: an image/diagram of the architecture would be very nice for a better overview

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+1. Convolutional Layers:
+- The input image is processed by the first convolutional layer, which applies 96 kernels of size 11x11x3 with a stride of 4 pixels.
+- The second convolutional layer applies 256 kernels of size 5x5x48.
+- Both convolutional layers are followed by ReLU (Rectified Linear Unit) activation and max-pooling layers.
+
+2.Fully Connected Layers:
+
+- After the convolutional layers, there are three fully connected layers.
+-The first two fully connected layers have 4096 neurons each, followed by dropout regularization to reduce overfitting.
+-The final fully connected layer has 1000 neurons, representing the 1000 object categories in the ImageNet dataset.
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:23]:
```suggestion
- The first two fully connected layers have 4096 neurons each, followed by dropout regularization to reduce overfitting.
- The final fully connected layer has 1000 neurons, representing the 1000 object categories in the ImageNet dataset.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+### Architechture of Alexnet
+
+The AlexNet architecture consists of the following layers:
+
+1. Convolutional Layers:
+- The input image is processed by the first convolutional layer, which applies 96 kernels of size 11x11x3 with a stride of 4 pixels.
+- The second convolutional layer applies 256 kernels of size 5x5x48.
+- Both convolutional layers are followed by ReLU (Rectified Linear Unit) activation and max-pooling layers.
+
+2.Fully Connected Layers:
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:19]:
```suggestion
2. Fully Connected Layers:
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+- After the convolutional layers, there are three fully connected layers.
+-The first two fully connected layers have 4096 neurons each, followed by dropout regularization to reduce overfitting.
+-The final fully connected layer has 1000 neurons, representing the 1000 object categories in the ImageNet dataset.
+
+3.Softmax Layer:
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:25]:
```suggestion
3. Softmax Layer:
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+
+- The output of the final fully connected layer is passed through a softmax activation function to produce the final class probabilities.
+
+4.Normalization Layers:
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:29]:
```suggestion
4. Normalization Layers:
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+
+- AlexNet also introduced Local Response Normalization (LRN) layers, which helped to improve the generalization performance of the network.
+
+5.ReLU Activation:
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:33]:
```suggestion
5. ReLU Activation:
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+- AlexNet was one of the first architectures to effectively utilize the Rectified Linear Unit (ReLU) activation function, which helped to address the vanishing gradient problem and accelerated the training process.
+
+### Key features
+
+- 'ReLU' is used as an activation function as opposed to 'tanh'
+    - 'tanh' was the standard activation function which was used back in those days.
+    - The primary advantage than 'ReLU' offered was the fact it reduced training time and also helped to alleviate the vanishing gradient problem
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:41]:
```suggestion
    - The primary advantage 'ReLU' offered, was the fact that it reduced training time and also helped to alleviate the vanishing gradient problem
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+- Usage of Dropout mechanism
+    - It is a regularization technique that helps to prevent overfitting in neural networks by randomly dropping out (setting to zero) a proportion of units (neurons) and their connections during the training process.
+    - Forces the network to learn more robust representation
+- Data Augmentation techniques such as flipping ,jittering ,cropping, etc. were performed.
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:45]:
```suggestion
- Data Augmentation techniques such as flipping, jittering, cropping, etc. were performed.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+    - The authors employed  extensive data augmentation techniques to artificially increase the size and diversity of the training data.
```
Comment (johko) [inline chapters/en/Unit 2 - Convolutional Neural Networks/alexnet.mdx:46]:
```suggestion
    - The authors employed extensive data augmentation techniques to artificially increase the size and diversity of the training data.
```

**Review by johko — COMMENTED** [review]
Thanks for the contribution, I just left some smaller nits, mainly spacing corrections :hugs:

**Comment by ash-01xor** [comment]
Had few build issues from my side , will resolve it and open a new PR for this.

**Pull request closed without merging**